src/components/instance-form-builder/GeneralInfoForm.jsx

```diff
@@ -172,21 +171,26 @@ class GeneralInfoForm extends React.Component {
             },
         ];
 
-        const saveAction = () => {
+        const saveAction = async () => {
             const formErrors = isFormValid(fields, this.formReference);
+            const { d2, instance } = this.props;
             if (formErrors.length > 0) {
                 this.props.snackbar.error(i18n.t("Please fix the issues before saving"));
                 return;
             }
+            const fieldKeys = fields.map(field => field.name);
+            const errorMessages = await getValidationMessages(d2, instance, fieldKeys);
 
-            this.setState({ isSaving: true });
-
-            this.props.originalInstance.remove(this.props.d2).then(() => {
-                this.props.instance.save(this.props.d2, this.props.isEdit).then(() => {
-                    this.setState({ isSaving: false });
-                    this.props.history.push("/instance-configurator");
+            if (!_(errorMessages).isEmpty()) {
```

Comment:
Chaining 1 method can be simplified: _.isEmpty(errorMessages)

Reply:
I am ok with both ways to write this kind of short lodash code. 

I think that it's acceptable -especially when using a non-chainable lodash method like `_.isEmpty`- because the OOP style is more declarative: `_(myList).isEmpty()`.

Reply:
Okay then! Will try to have it in mind to keep the same coding style

Reply:
Yep, but feel free to use `_.someMethod(var)` whenever it makes sense to you, IMO it's no problem to have those minor nuances in the code :)